pilot/pkg/xds/nds.go

```diff
@@ -61,20 +61,8 @@ var skippedNdsConfigs = sets.New(
 )
 
 func ndsNeedsPush(req *model.PushRequest, proxy *model.Proxy) bool {
-	if proxy.Type == model.Ztunnel {
-		// Not supported for ztunnel
-		return false
-	}
-	if req == nil {
-		return true
-	}
-	if !req.Full {
-		// NDS generally handles full push. We only allow partial pushes, when headless endpoints change.
-		return headlessEndpointsUpdated(req)
```

Comment:
seems req.Full is always true when we set `HeadlessEndpointUpdate`

Reply:
I think it was changed in https://github.com/istio/istio/pull/49463 - Should this condition be removed now?

Reply:
I think it was changed in https://github.com/istio/istio/pull/49463 - Should this condition be removed now?

Reply:
I think so